src/main/groovy/edu/oregonstate/mist/invencycle/resources/BikeResource.groovy

```diff
@@ -76,47 +75,28 @@ class BikeResource extends Resource {
     @Produces(MediaType.APPLICATION_JSON)
     public Response postBike (@Valid Bike newBike) {
         Response returnResponse
-        try {
-            bikeDAO.postBrakeFront(newBike.brake_make_front,
-                                   newBike.brake_model_front,
-                                   newBike.brake_rotor_size_front)
-
-            bikeDAO.postBrakeRear(newBike.brake_make_rear,
-                                  newBike.brake_model_rear,
-                                  newBike.brake_rotor_size_rear)
-            bikeDAO.postDerailuerFront(newBike.derailuer_make_front,
-                                       newBike.derailuer_model_front,
-                                       newBike.derailuer_speeds_front)
-            bikeDAO.postDerailuerRear(newBike.derailuer_make_rear,
-                                      newBike.derailuer_model_rear,
-                                      newBike.derailuer_speeds_rear)
-            bikeDAO.postFrameSize(newBike.frame_size_name,
-                                  newBike.frame_size_cm)
-            bikeDAO.postFork(newBike.fork_make,
-                             newBike.fork_model,
-                             newBike.fork_travel)
-            bikeDAO.postShock(newBike.shock_make,
-                              newBike.shock_model,
-                              newBike.shock_travel)
-            bikeDAO.postWheelFront(newBike.wheel_size_front,
-                                   newBike.rim_make_front,
-                                   newBike.rim_model_front,
-                                   newBike.hub_make_front,
-                                   newBike.hub_model_front)
-            bikeDAO.postWheelRear(newBike.wheel_size_rear,
-                                  newBike.rim_make_rear,
-                                  newBike.rim_model_rear,
-                                  newBike.hub_make_rear,
-                                  newBike.hub_model_rear)
-            bikeDAO.postBike(newBike.make,
-                             newBike.model,
-                             newBike.msrp,
-                             newBike.year,
-                             newBike.type)
 
+        try {
+            addBike(newBike)
             returnResponse = Response.ok("Bike successfully created").build()
         } catch (Exception e) {
-            returnResponse = internalServerError("Internal server error").build()
+            String dbError = e.cause.toString()
+
+            if (dbError.contains("cannot insert NULL into (\"MISTSTU3\".\"BIKE\".\"BIKE_BIKE_TYPE_ID\")")) {
```

Comment:
This set of if & else if statements are a bit repetitive. How about using something like:
```
validationErrorMsg = getDataError(dbError)
if (!validationErrorMsg) {
    returnResponse = badRequest(validationErrorMsg).build()
} else {
    logger.error("Exception while calling postBike", e)
    returnResponse = internalServerError("Internal server error").build()
} ....

getDataError(dbError) {
    def errorMessages = ["bike_make", "bike_model", "bike_year", "bike_msrp"]
    def frienldyMessage

    errorMessages.each {
        if (dbError.contains("cannot insert NULL into (\"MISTSTU3\".\"BIKE\".\"${it.toUpperCase()}\")") {
            fiendlyMessage = "${it} cannot be NULL"
        }
    }

    fiendlyMessage
}
```

I didn't compile the pseudocode above so there are likely some typos. This allows you to have a separate private method that does the validation for you and returns an error message. It allows your resource method to just focus on the handling and routing of your request to smaller more focused methods.